Backend/src/API/REST/routes/team/teamRouter.ts

```diff
@@ -120,4 +121,220 @@ const invite_person_to_a_team = (): Router => {
     return router;
 }
 
-export { create_team, invite_person_to_a_team }
+const add_user_to_admin = (): Router => {
+    let router = express.Router();
+
+    router.put("/teams/:id/admin", check_auth(), async (req: any, res) => {
+        try {
+            const {userId}: {userId: ObjectId}  = req.body;
+            const teamId: string = req.params.id;
+
+            if(!(userId && teamId))
+                return res.status(400).send("Missing parameters.");
+
+            const user = await UserService.getUserById(userId)
+            if (! user )
+                return res.status(400).send("User does not exist.");
+
+            const team = await TeamService.getTeamById(teamId);
+            if (! team)
+                return res.status(400).send("Team does not exist.");
+
+            var tempUserId = String(userId);
+            var tempListAdmins = team.adminIds.map(String);
+            var tempListMembers = team.memberIds.map(String);
+
+            if (tempListAdmins.includes(tempUserId))
+                return res.status(409).send("User is already an admin.")
+
+            if (!tempListMembers.includes(tempUserId))
+                return res.status(409).send("User is not a member of the team. It should be first a member to become an admin.")            
+
+            try {
+                const team_updated = await TeamService.addAdminToTeam(teamId, userId);
+
+                if (!team_updated)
+                    return res.status(400).send("Error when changing to user to admin profile.");
+
+                return res.status(200).json("User has been changed to admin.");
+
+            } catch(err) {
+                console.error("Error when trying to register new admin of a given a team.")
+                console.error(JSON.stringify(err));
+                console.error(err);
+                return res.status(500).send("Unable to register new admin.");
+            }
+        } catch(e) {
+            /* Added since a test proved that if user sends a request with incorrect parameter names, it is able to shutdown the server. */
+            console.error("Error in add_user_to_admin()")
+            console.error(JSON.stringify(e));
+            console.error(e);
+            return res.status(500).send("Internal error.");
+        }
+
+    })
+
+    return router;
+}
+
+const join_member = (): Router => {
+    let router = express.Router();
+
+    router.put("/teams/:id/join", check_auth(), async (req: any, res) => {
+        try {
+            const {userId}: {userId: ObjectId}  = req.body;
```

Comment:
Question: The userId appears to be someone else, than the user who sends the request. Should a user be able to add another user to a team? As it is at the moment, anyone can add anybody to any team.
I would suggest, that a user calls this route if they want to join the specified team, and joining is only allowed if they are in the invitation list.